chanbackup/backupfile_test.go

```diff
@@ -45,160 +45,217 @@ func assertFileDeleted(t *testing.T, filePath string) {
-			t.Fatalf("#%v, unable to swap file: %v", i, err)
+				// TODO(roasbeef): mock out fs calls?
+			}
 
-		// If this is an invalid test case, and we passed it, then
-		// we'll return an error.
-		case err == nil && !testCase.valid:
-			t.Fatalf("#%v file swap should have failed: %v", i, err)
-		}
+			// With our backup created, we'll now attempt to swap
+			// out this backup, for the old one.
+			err = backupFile.UpdateAndSwap(newPackedMulti)
+			require.NoError(t, err)
 
-		if !testCase.valid {
-			continue
-		}
+			// If we read out the file on disk, then it should match
+			// exactly what we wrote. The temp backup file should
+			// also be gone.
+			assertBackupMatches(t, fileName, newPackedMulti)
+			assertFileDeleted(t, tempFileName)
 
-		// If we read out the file on disk, then it should match
-		// exactly what we wrote. The temp backup file should also be
-		// gone.
-		assertBackupMatches(t, testCase.fileName, newPackedMulti)
-		assertFileDeleted(t, testCase.tempFileName)
+			// Now that we know this is a valid test case, we'll
+			// make a new packed multi to swap out this current one.
+			newPackedMulti2, err := makeFakePackedMulti()
+			require.NoError(t, err)
 
-		// Now that we know this is a valid test case, we'll make a new
-		// packed multi to swap out this current one.
-		newPackedMulti2, err := makeFakePackedMulti()
-		if err != nil {
-			t.Fatalf("unable to make test backup: %v", err)
-		}
+			// We'll then attempt to swap the old version for this
+			// new one.
+			err = backupFile.UpdateAndSwap(newPackedMulti2)
+			require.NoError(t, err)
 
-		// We'll then attempt to swap the old version for this new one.
-		err = backupFile.UpdateAndSwap(PackedMulti(newPackedMulti2))
-		if err != nil {
-			t.Fatalf("unable to swap file: %v", err)
-		}
+			// Once again, the file written on disk should have been
+			// properly swapped out with the new instance.
+			assertBackupMatches(t, fileName, newPackedMulti2)
 
-		// Once again, the file written on disk should have been
-		// properly swapped out with the new instance.
-		assertBackupMatches(t, testCase.fileName, newPackedMulti2)
-
-		// Additionally, we shouldn't be able to find the temp backup
-		// file on disk, as it should be deleted each time.
-		assertFileDeleted(t, testCase.tempFileName)
-
-		// Now check if archive was created when noBackupArchive is
-		// false.
-		archiveDir := filepath.Join(
-			filepath.Dir(testCase.fileName),
-			DefaultChanBackupArchiveDirName,
-		)
-		if !testCase.noBackupArchive {
+			// Additionally, we shouldn't be able to find the temp
+			// backup file on disk, as it should be deleted each
+			// time.
+			assertFileDeleted(t, tempFileName)
+
+			// Now check if archive was created when noBackupArchive
+			// is false.
+			archiveDir := filepath.Join(
+				filepath.Dir(fileName),
+				DefaultChanBackupArchiveDirName,
+			)
 			files, err := os.ReadDir(archiveDir)
 			require.NoError(t, err)
 			require.Len(t, files, 1)
```

Comment:
Since the two archives share the same filename most of the time, this would pass, otherwise it will fail.